File: CCSIG/21w3 problemset/hammingDistanceSum.py
import operator

a : str = "001"
b : str = "1111111111111111"*200000

totalSum : int = 0


def hammingDistanceSum(a: str, b:str) -> int:
    a: list[int] = list(map(int, a))
    b: list[int] = list(map(int, b))


    sumOF:int = sum(map(operator.xor, a, b))

    return sumOF




for i in range(0, (len(b)//len(a))*len(a)):
    if i + len(a) > len(b):
        break
    totalSum += hammingDistanceSum(a, b[i:i+len(a)])

print(totalSum)

File: CCSIG/21w3 problemset/hammingDistanceSum.cpp
#include <iostream>
#include <string>
#include <vector>
#include <numeric> // for accumulate
#include <functional> // for std::bit_xor

int hammingDistanceSum(const std::string& a, const std::string& b) {
    int sum = 0;
    for (size_t i = 0; i < a.size(); ++i) {
        int bitA = a[i] - '0';
        int bitB = b[i] - '0';
        sum += bitA ^ bitB; // XOR operation
    }
    return sum;
}

int main() {
    std::string a = "001";
    std::string b = std::string(16 * 200000, '1'); // 16 * 200000 = 3.2 million '1's

    int totalSum = 0;
    size_t chunkSize = a.size();

    for (size_t i = 0; i + chunkSize <= b.size(); i += 1) {
        totalSum += hammingDistanceSum(a, b.substr(i, chunkSize));
    }

    std::cout << totalSum << std::endl;
    return 0;
}
